plugins/azure/fix_plugin_azure/resource/compute.py

```diff
@@ -726,7 +1123,62 @@ def collect_usage_metrics(
 
         update_resource_metrics(volumes, metric_result, metric_normalizers)
 
+    @staticmethod
+    def _get_nearest_size(size: int, lookup_map: Dict[int, Any]) -> int:
+        list_sizes = list(lookup_map.keys())
+        target = size
+
+        return min(list_sizes, key=lambda x: abs(x - target))
+
+    def post_process(self, graph_builder: GraphBuilder, source: Json) -> None:
+        if (
+            (volume_type := self.volume_type)
+            and (location := self.location)
+            and (size := self.volume_size)
+            and (iops := self.volume_iops)
+            and (throughput := self.volume_throughput)
+        ):
+            if volume_type not in ["UltraSSD_LRS", "PremiumV2_LRS"]:
+                return
+
+            generic_size = AzureDiskType.create_custom_size(location, volume_type, size, iops, throughput)
+            AzureDiskType.collect(generic_size, graph_builder)
```

Comment:
If I have multiple ultra or premium SSD disks with same properties, this will create multiple disk sizes, but should be only one?

Reply:
Fixed, thanks